rundeckapp/grails-app/services/rundeck/services/scm/ScmLoaderService.groovy

```diff
@@ -115,18 +117,33 @@ class ScmLoaderService implements EventBusAware {
                         scmPluginMeta.put(projectIntegration, pluginConfigData)
                     }
                     if(pluginConfigData && pluginConfigData.enabled){
-                        try {
-                            if (integration == scmService.EXPORT) {
-                                processScmExportLoader(project, pluginConfigData, state)
-                            }else{
-                                processScmImportLoader(project, pluginConfigData, state)
-                            }
+                        boolean process = false
+                        int retryCount = 0
+                        long retryTimes = getScmLoaderInitialRetryTimes()
+                        long retryDelay = getScmLoaderInitialRetryDelay()
+
+                        while (!process){
+                            try {
+                                if (integration == scmService.EXPORT) {
+                                    processScmExportLoader(project, pluginConfigData, state)
+                                }else{
+                                    processScmImportLoader(project, pluginConfigData, state)
+                                }
+                                process = true
 
-                        } catch (Throwable t) {
-                            log.error("processMessages error: $project/$integration: ${t.message}")
-                            scmFailedProjectInit.put(projectIntegration, pluginConfigData)
-                            removingLoaderProcess(project, integration)
+                            } catch (Throwable t) {
+                                retryCount++
+                                log.error("processMessages error: $project/$integration: ${t.message} retrying ${retryCount}/${retryTimes}")
+                                Thread.sleep(retryDelay)
```

Comment:
the retry delay is fixed here, but maybe it should back off like: `Thread.sleep(retryDelay*retryCount)`